src/setuptools_scm/hg.py

```diff
@@ -68,15 +70,23 @@ def parse(root, config=None):
         pass  # unpacking failed, old hg
 
 
-def get_latest_normalizable_tag(root):
-    # Gets all tags containing a '.' (see #229) from oldest to newest
+def get_latest_normalizable_tag(config):
+    """Return the most recent tag that matches tag_regex
+
+    """
+    # Get all tags containing a period, then filter using tag_regex.
+    root = config.absolute_root
     cmd = [
         "hg", "log", "-r", "ancestors(.) and tag('re:\\.')", "--template", "{tags}\n"
     ]
     outlines = do(cmd, root).split()
     if not outlines:
         return "null"
-    tag = outlines[-1].split()[-1]
+    tag_regex = re.compile(config.tag_regex)
+    try:
+        tag = next(v for v in reversed(outlines) if tag_regex.match(v))
```

Comment:
please use a list comp before the `if not outlines` line to filter using the regex instead